contracts/Erc20Quest.sol

```diff
@@ -15,28 +20,47 @@ contract Erc20Quest is Quest {
         string memory allowList_,
         uint256 rewardAmountInWeiOrTokenId_,
         string memory questId_,
-        address receiptContractAddress_
+        address receiptContractAddress_,
+        uint256 questFee_,
+        address protocolFeeRecipient_,
+        address factoryContractAddress_
     )
-        Quest(
-            rewardTokenAddress_,
-            endTime_,
-            startTime_,
-            totalAmount_,
-            allowList_,
-            rewardAmountInWeiOrTokenId_,
-            questId_,
-            receiptContractAddress_
-        )
-    {}
+    Quest(
+        rewardTokenAddress_,
+        endTime_,
+        startTime_,
+        totalAmount_,
+        allowList_,
+        rewardAmountInWeiOrTokenId_,
+        questId_,
+        receiptContractAddress_
+    ) {
+        questFee = questFee_;
+        totalRedeemers = rewardAmountInWeiOrTokenId / totalAmount;
+        protocolFeeRecipient = protocolFeeRecipient_;
+        questFactoryContract = QuestFactory(factoryContractAddress_);
+        factoryContractAddress = factoryContractAddress_;
+    }
 
     function start() public override {
-        if (IERC20(rewardToken).balanceOf(address(this)) < totalAmount) revert TotalAmountExceedsBalance();
+        if (IERC20(rewardToken).balanceOf(address(this)) < (totalAmount + (totalAmount * questFee / 10_000))) revert TotalAmountExceedsBalance();
         super.start();
     }
 
+    function mintReceipt(uint amount_, string memory questId_, bytes32 hash_, bytes memory signature_) public {
+        if (hasStarted == false) revert NotStarted();
+        if (isPaused == true) revert QuestPaused();
+        if (block.timestamp < startTime) revert NotStarted();
+        if (block.timestamp > endTime) revert QuestEnded();
```

Comment:
Do we want the `QuestEnded` check on Erc1155Quest `claim` function?

Reply:
Since we're the only ones running the 1155 Quests I'm not _as_ concerned about it.